R/misc.R

```diff
@@ -98,16 +98,17 @@ add_class <- function(x, cls) {
   x
 }
 
-strata_check <- function(strata, data) {
+check_strata <- function(strata, data, call = caller_env()) {
+  check_string(strata, allow_null = TRUE, call = call)
+
   if (!is.null(strata)) {
-    if (!is.character(strata) | length(strata) != 1) {
-      cli_abort("{.arg strata} should be a single name or character value.")
-    }
     if (inherits(data[, strata], "Surv")) {
-      cli_abort("{.arg strata} cannot be a {.cls Surv} object. Use the time or event variable directly.")
-    }
-    if (!(strata %in% names(data))) {
-      cli_abort("{strata} is not in {.arg data}.")
+      cli_abort(c(
+        "{.arg strata} cannot be a {.cls Surv} object.",
```

Comment:
Technically `strata` is a character or name. but it is used to refer to column in the data set. So I think `.field` would be better here as it refers to the value

```suggestion
        "{.field strata} cannot be a {.cls Surv} object.",
```

Reply:
Good point, thank you!